spring-integration-core/src/main/java/org/springframework/integration/config/AggregatorFactoryBean.java

```diff
@@ -48,128 +49,267 @@
-		this.aggregator.setReleaseStrategy(releaseStrategy);
+		this.releaseStrategy = releaseStrategy;
 	}
 
 	public void setGroupTimeoutExpression(Expression groupTimeoutExpression) {
-		this.aggregator.setGroupTimeoutExpression(groupTimeoutExpression);
+		this.groupTimeoutExpression = groupTimeoutExpression;
 	}
 
 	public void setForceReleaseAdviceChain(List<Advice> forceReleaseAdviceChain) {
-		this.aggregator.setForceReleaseAdviceChain(forceReleaseAdviceChain);
+		this.forceReleaseAdviceChain = forceReleaseAdviceChain;
 	}
 
 	public void setTaskScheduler(TaskScheduler taskScheduler) {
-		this.aggregator.setTaskScheduler(taskScheduler);
-	}
-
-	@Override
-	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
-		this.aggregator.setApplicationEventPublisher(applicationEventPublisher);
+		this.taskScheduler = taskScheduler;
 	}
 
 	public void setDiscardChannel(MessageChannel discardChannel) {
-		this.aggregator.setDiscardChannel(discardChannel);
+		this.discardChannel = discardChannel;
 	}
 
 	public void setDiscardChannelName(String discardChannelName) {
-		this.aggregator.setDiscardChannelName(discardChannelName);
-	}
-
-	public void setSendPartialResultOnExpiry(boolean sendPartialResultOnExpiry) {
-		this.aggregator.setSendPartialResultOnExpiry(sendPartialResultOnExpiry);
+		this.discardChannelName = discardChannelName;
 	}
 
-	public void setMinimumTimeoutForEmptyGroups(long minimumTimeoutForEmptyGroups) {
-		this.aggregator.setMinimumTimeoutForEmptyGroups(minimumTimeoutForEmptyGroups);
+	public void setSendPartialResultOnExpiry(Boolean sendPartialResultOnExpiry) {
+		this.sendPartialResultOnExpiry = sendPartialResultOnExpiry;
 	}
 
-	public void setReleasePartialSequences(boolean releasePartialSequences) {
-		this.aggregator.setReleasePartialSequences(releasePartialSequences);
+	public void setMinimumTimeoutForEmptyGroups(Long minimumTimeoutForEmptyGroups) {
+		this.minimumTimeoutForEmptyGroups = minimumTimeoutForEmptyGroups;
 	}
 
-	public void setExpireGroupsUponTimeout(boolean expireGroupsUponTimeout) {
-		this.aggregator.setExpireGroupsUponTimeout(expireGroupsUponTimeout);
+	public void setExpireGroupsUponTimeout(Boolean expireGroupsUponTimeout) {
+		this.expireGroupsUponTimeout = expireGroupsUponTimeout;
 	}
 
 	@Override
 	protected AggregatingMessageHandler createHandler() {
+		MessageGroupProcessor outputProcessor;
+		if (this.processorBean instanceof MessageGroupProcessor) {
+			outputProcessor = (MessageGroupProcessor) this.processorBean;
+		}
+		else {
+			if (this.methodName == null) {
+				outputProcessor = new MethodInvokingMessageGroupProcessor(this.processorBean);
+			}
+			else {
+				outputProcessor = new MethodInvokingMessageGroupProcessor(this.processorBean, this.methodName);
+			}
+		}
+		this.aggregator = new AggregatingMessageHandler(outputProcessor);
+
+		if (this.applicationContext != null) {
+			this.aggregator.setApplicationContext(this.applicationContext);
```

Comment:
Yes, I noticed that, but I would be concerned about doing that in a point release.